### DISCUSSION TITLE
Inquiry Regarding Spherical Harmonics Function in MOOSE

### DISCUSSION THREAD

**Discussion by hityyds** [idaholab/moose#27181] [Q&A General] [answered]
Dear Community,

I am currently working on a program that requires the use of spherical harmonics functions, and I am wondering if there is any relevant code available in MOOSE. I am aware that MOOSE incorporates Legendre polynomial functions, which led me to inquire about the availability of spherical harmonics functions as well.

I have attempted to write my own code for spherical harmonics functions. However, I consistently encounter peculiar results during program execution.  Given that my mathematical background is not very strong, I am concerned that my understanding of the spherical harmonics function formulas may be incorrect.

I believe having access to pre-implemented spherical harmonics functions within MOOSE would significantly facilitate my project's progress. Therefore, I would greatly appreciate it if anyone could confirm the availability of such functionality within MOOSE.

Thank you very much for your assistance.

Sincerely,
hityyds

**Comment by GiudGiud** [comment, marked as answer]
Hello

I dont think we do, but we have the architecture for implementing more basis here:
https://github.com/idaholab/moose/tree/next/modules/functional_expansion_tools/src/series
and re-using that to implemenr spherical harmonics will make things easier

Guillaume

**Reply by hityyds** [reply]
Thank you very much for your response and I appreciate your guidance. I will explore the provided resources and consider how to adapt them for implementing spherical harmonics in my program.

**Reply by GiudGiud** [reply]
took another look, looks like spherical harmonics were in the todo list for the FET module there
https://mooseframework.inl.gov/modules/functional_expansion_tools/index.html#1fbf33b7-3aa7-48c5-a31b-1cde80d659ec

if you get an implementation working, I would love to get it into MOOSE after
